backend/tests/sync_status_integration_test.go

```diff
@@ -0,0 +1,635 @@
+package integration
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	catalogclient "github.com/doron-cohen/argus/backend/api/client"
+	"github.com/doron-cohen/argus/backend/internal/server"
+	"github.com/doron-cohen/argus/backend/sync"
+	syncclient "github.com/doron-cohen/argus/backend/sync/api/client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestSyncStatusComponentsCountBasic tests the basic functionality of componentsCount reporting
+func TestSyncStatusComponentsCountBasic(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test in short mode")
+	}
+
+	// Clear database before test
+	clearDatabase(t)
+
+	testDataPath := getTestDataPath(t)
+
+	// Create config with filesystem source pointing to testdata
+	testConfig := TestConfig
+	fsConfig := sync.NewFilesystemSourceConfig(testDataPath, 1*time.Second)
+	testConfig.Sync = sync.Config{
+		Sources: []sync.SourceConfig{
+			sync.NewSourceConfig(fsConfig.GetConfig()),
+		},
+	}
+
+	// Start server with sync enabled
+	stop, err := server.Start(testConfig)
+	require.NoError(t, err)
+	defer stop()
+
+	// Wait for server to start
+	time.Sleep(2 * time.Second)
```

Comment:
There is a utility to wait for server health to become OK. I think it is in utils.go